Add Terraform Cloud integration docs (#30102)
hashicorp/terraform: website/docs/language/settings/backends/azurerm.html.md

# azurerm

Stores the state as a Blob with the given Key within the Blob Container within [the Blob Storage Account](https://docs.microsoft.com/en-us/azure/storage/common/storage-introduction).

> This backend supports state locking and consistency checking via native capabilities of Azure Blob Storage.

Reviewer: ```suggestion
This backend supports state locking and consistency checking with Azure Blob Storage native capabilities.
```
